app/helpers/categories_helper.rb

```diff
@@ -0,0 +1,34 @@
+module CategoriesHelper
+  def save_category(category)
```

Comment:
i see your thought process here, but i don't think keeping this logic (for saving, deleting, and updating) in the controller is a bad idea. it's definitely easier to read if it's all the controller. and it's functionality that's key to controller functionality